linux_os/guide/system/accounts/accounts-restrictions/account_expiration/ensure_shadow_group_empty/oval/shared.xml

```diff
@@ -0,0 +1,48 @@
+<def-group>
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    {{{ oval_metadata("Ensure shadow group is empty") }}}
+    <criteria>
+      <criterion test_ref="tst_shadow_group_empty" comment="shadow group is empty" />
+      <criterion test_ref="tst_no_user_assigned_shadow_group" comment="no user is assigned to the shadow group" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="only_one_exists"
```

Comment:
If the system has no `shadow` group. For example, if the `root` group is used instead, this rule will fail.
The intention of the rule is to ensure the `shadow` group is empty. If the `shadow` group does not exist, the rule should pass. Maybe using `check_existence="any_exist"` should work.

Reply:
it this a valid case though?
If you want shadowed passwords I believe it might be mandatory to have the shadow group.
might need to dig some man pages about this

Reply:
I couldn't check the sle and ubuntu distros, but in RHEL, for example, there is no `shadow` group. The `/etc/shadow` is group-owned by `root` in this case. My concern is more about the robustness of the rule if anything change in the OS, for example.